src/shell/shell_cmds.cpp

```diff
@@ -1450,6 +1450,21 @@ void DOS_Shell::CMD_TIME(char * args) {
 		mem_writed(BIOS_TIMER, static_cast<uint32_t>(ticks_now));
 		return;
 	}
+	uint32_t newhour, newminute, newsecond;
+	if (sscanf(args, "%u:%u:%u", &newhour, &newminute, &newsecond) == 3) {
+		if (newhour > 23 || newminute > 59 || newsecond > 59)
+			WriteOut(MSG_Get("SHELL_CMD_TIME_ERROR"));
+		else {
+			reg_ch = static_cast<Bit8u>(newhour);
+			reg_cl = static_cast<Bit8u>(newminute);
+			reg_dh = static_cast<Bit8u>(newsecond);
```

Comment:
Suggest using `uint8_t` (we're trying to eventually use cstdint's and no longer need the Bit* wrapper types).

Reply:
This part was originally modelled on the manual setting of DATE command. Now I changed Bit8u to uint8_t in both DATE and TIME commands.